apps/prairielearn/elements/pl-file-upload/pl-file-upload.js

```diff
@@ -83,26 +105,33 @@
         accept: (file, done) => {
           // fuzzy case match
           const fileNameLowerCase = file.name.toLowerCase();
-          if (this.acceptedFilesLowerCase.includes(fileNameLowerCase)) {
+          if (this.findAcceptedFileName(fileNameLowerCase)) {
             return done();
           }
           return done('invalid file');
         },
         addedfile: (file) => {
+          const existingFileSize = this.files.reduce((prev, next) => prev + next.size, 0);
+          if (existingFileSize + file.size > this.maxFileSizeMB * 1024 * 1024) {
```

Comment:
The current maxFileSizeMB is set to 5 MB, so it already has a fairly generous buffer to the 10 MB supported by the server that factors in the overhead. I haven't really changed anything about this threshold, other than making it easier to update in case the 10 MB limit ever changes or someone manually wants to change it for their self-hosted instance.

Reply:
Ah, just goes to show that I don't know what the actual limit is 😂 thanks for explaining, I agree this is fine.